src/WorkItemMigrator/JiraExport/JiraItem.cs

```diff
@@ -47,13 +47,9 @@ private static List<JiraRevision> BuildRevisions(JiraItem jiraItem, IJiraProvide
             if (createdOn == DateTime.MinValue)
                 Logger.Log(LogLevel.Debug, "created key was not found, using DateTime default value");
 
-
-            var changelog = jiraProvider.DownloadChangelog(issueKey).ToList();
+            var changelog = jiraProvider.DownloadChangelog(issueKey).OrderByDescending(c => (long)c.SelectToken("$.id")).ToList();
```

Comment:
Are we sure that this explicit cast won't throw exceptions?

Reply:
Pretty much, this Id is returned from the database. Older schemas use BIGINT which maps to long. Newer schema says it is numeric(18,0), that should also be fine in a long. See: https://developer.atlassian.com/server/jira/platform/database-schema/